### ISSUE TITLE
Measuring Accuracy During Training

### ISSUE THREAD

**Issue by kyunghoyu** [clip-italian/clip-italian#22]
Hello? I have written a code to measure accuracy based on the method you provided. Does the code look correct for measuring accuracy?

```
def train_step(state, batch):
        dropout_rng, new_dropout_rng = jax.random.split(state.dropout_rng)

        def compute_loss(params):
            logits = state.apply_fn(
                **batch, params=params, dropout_rng=dropout_rng, train=True
            )[0]
            loss = clip_loss(logits)
            return loss, logits
        
        grad_fn = jax.value_and_grad(compute_loss, has_aux=True)
        (loss, logits), grad = grad_fn(state.params)
        grad = jax.lax.pmean(grad, "batch")

        new_state = state.apply_gradients(grads=grad, dropout_rng=new_dropout_rng)

        ground_truth = jnp.arange(len(logits))
        acc_i = ((jnp.argmax(logits, 1) == ground_truth)).mean()
        acc_t = ((jnp.argmax(logits, 1) == ground_truth)).mean()
        acc = (acc_i + acc_t) / 2
        
        metrics = {
            "acc" : acc,
            "loss": loss,
            "learning_rate": decay_lr_schedule_fn(state.step),
        }
        metrics = jax.lax.pmean(metrics, axis_name="batch")

        return new_state, metrics
```

```
def eval_step(params, batch):
        output = model(**batch, params=params, train=False)
        #logits_per_image, logits_per_text, text_embeds, image_embeds, text_outputs, vision_outputs
        logits = output[0] #logits
        loss = clip_loss(logits)
        
        ground_truth = jnp.arange(len(logits))
        acc_i = ((jnp.argmax(logits, 1) == ground_truth)).mean()
        acc_t = ((jnp.argmax(logits, 1) == ground_truth)).mean()
        acc = (acc_i + acc_t) / 2
        # summarize metrics
        metrics = {
            "acc" : acc,
            "loss": loss,
            }
        metrics = jax.lax.pmean(metrics, axis_name="batch")
        return metrics
```
